pkg/common/common.go

```diff
@@ -246,3 +248,22 @@ func AreSocketsEqual(socketA string, socketB string) (bool, error) {
 
 	return (hostAIP.Equal(hostBIP) && (portA == portB)), nil
 }
+
+func StrValueFromMapOrDefault(configMap map[string]any, param string, defaultVal string) (string, error) {
```

Comment:
There are some similar functions in https://github.com/project-zot/zot/blob/main/pkg/api/config/redis/redis.go#L249.
To be more specific the [cast ](https://github.com/spf13/cast) library which is also used internally by the library reading the config file (https://github.com/spf13/viper): https://github.com/project-zot/zot/blob/main/pkg/api/config/redis/redis.go#L254C1-L255C30